File: articles/bounded-rationality/r/run_all_bounded_rationality_workflows.R
# run_all_bounded_rationality_workflows.R
# Run all R workflows for Bounded Rationality.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "bounded_rationality_workflow.R",
  "satisficing_profiles.R",
  "search_cost_tables.R",
  "adaptive_aspiration_reports.R",
  "domain_constraint_diagnostics.R",
  "bounded_rationality_review_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R bounded rationality workflows completed.")

File: articles/bounded-rationality/python/run_all_bounded_rationality_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Bounded Rationality."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "satisficing_search_model.py",
    "optimizing_vs_satisficing.py",
    "aspiration_adaptation.py",
    "search_cost_diagnostics.py",
    "organizational_constraint_model.py",
    "stopping_rule_analysis.py",
    "decision_record_exporter.py",
    "bounded_rationality_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python bounded rationality workflows completed.")


if __name__ == "__main__":
    main()
